src/Temporalio/Client/WorkflowStartOptions.cs

```diff
@@ -69,14 +69,14 @@ public WorkflowStartOptions(string id, string taskQueue)
         public string? CronSchedule { get; set; }
 
         /// <summary>
-        /// Gets or sets the memo for the workflow.
+        /// Gets or sets the memo for the workflow. Values for the memo cannot be null.
         /// </summary>
         public IReadOnlyCollection<KeyValuePair<string, object>>? Memo { get; set; }
 
         /// <summary>
         /// Gets or sets the search attributes for the workflow.
         /// </summary>
-        public IReadOnlyCollection<KeyValuePair<string, object>>? SearchAttributes { get; set; }
+        public SearchAttributeCollection? TypedSearchAttributes { get; set; }
```

Comment:
Do you want to call this `TypedSearchAttributes` for uniformity across SDKs? Since we never had untyped ones I can see an argument for just calling this `SearchAttributes`.

Reply:
Exactly. I accept that this could be better in .NET but is best to be named the same in every language.